states/wyoming/parse_text.py

```diff
@@ -0,0 +1,64 @@
+import json
+from common import dir_path
+
+# link to data: https://sos.wyo.gov/Elections/Docs/WYCountyClerks_AbsRequest_VRChange.pdf 
+
+def get_county_dict():
+  county = {
+    'local': '',
+    'official': '',
+    'emails': [],
+    'faxes': [],
+    'phones': [],
+    'county': '',
+    'party': ''
+  }
+  return county
+
+
+def generate_county_dict_list(lines):
+  counties = []
+
+  new_county = True
+  county = get_county_dict()
+  for line in lines:
+      line = line.strip(' \n')
+      if 'County' in line:
+        if not new_county:
+          counties.append(county)
+          new_county = True
+          county = get_county_dict()
+        county['locale'] = line[:-len(' Clerk')]
+        county['county'] = line[:-len(' Clerk')]
+        new_county = False
+      elif '@' in line:
+        if line.startswith('Email: '):
+          line =  line[len('Email: '):]
+        county['emails'].append(line)
+      elif 'Ph.' in line:
+        county['phones'].append(line.strip('Ph. ').replace('.', '-'))
+      elif 'Fax' in line:
+        county['faxes'].append(line.strip('Fax ').replace('.', '-'))
+      else:
+        if len(line) > 3:
+          if 'address' not in county:
+            county['address'] = ''
+          county['address'] += (line + ' ')
+
+  return counties
+
+
+def main():
```

Comment:
Can you make the code generate everything from scratch?  Ideally, it would download the relevant files (into a gitignore'd cache) and just write the results out directly storing any intermediate forms into the gitignore'd cache.  Also, please use `cache_request` so that we can overwrite the cache when we need to rerun everything.  We want to be able to rerun everything once a week de novo.

Reply:
The problem is that this is the raw text:

 Page 
1 of 
3 Albany County Clerk
 Big Horn County Clerk
 525 Grand Avenue, Suite 202
 Laramie, WY 82070
 Ph. 307.721.2546
 Fax 307.721.2544
 Email: 
elections@co.albany.wy.us
 County Courthouse, P.O. Box 31
 Basin, WY 82410
-0031
 Ph. 307.568.2357
 Fax 307.568.9375
 Email: 
Lori.Smallwood@bighorncountywy.gov
           
annette.dillon@bighorncountywy.gov
 Campbell County Clerk
 Carbon County Clerk
 P.O. Box 3010
 Gillette, WY 82717
-3010
 Ph. 307.686.1892
 Fax 307.687.6455
 Email: 
cds02@ccgov.net
 P.O. Box 6
 Rawlins, WY 82301
-0006
 Ph. 307.328.2650
 Fax 307.328.2669
 Email: 
elections@carbonwy.com
     Converse County Clerk
 Crook County Clerk
 107 North 5th Street, Suite 114
 Douglas, WY 82633
-2448 Ph. 307.358.2244
 Fax 307.358.5998
 Email:
janeen.morvee@conversecountywy.gov
            
 P.O. Box 37
 Sundance, WY 82729
-0037
 Ph. 307.283.1323
 Fax 307.283.3038
 Email: 
LindaF@crookcounty.wy.gov
                
melissaj@crookcounty.wy.gov
 Fremont County Clerk
 Goshen County Clerk
 450 North 2nd Street, Room 220
 Lander, WY 82520
-0090
 Ph. 307.332.1089
 Fax 307.332.1132
 Email: 
elections@fremontcountywy.gov
 P.O. Box 160
 Torrington, WY 82240
-0160
 Ph. 307.532.4051
 Fax 307.532.7375
 Email: 
countyclerk@goshencounty.org
 
 Page 
2 of 
3  Hot Springs County Clerk
 Johnson County Clerk
 415 Arapahoe
 Thermopolis, WY 82443
-2783 Ph. 307.864.3515
 Fax 307.864.3333
 Email: 
elections
@hscounty.com
 76 North Main Street
 Buffalo, WY 82834
-1883
 Ph. 307.684.7272
 Fax 307.684.2708
 Email: 
vedelman@johnsoncowy.us
    clerk@johnsoncowy.us
 Laramie County Clerk
 Lincoln County Clerk
 P.O. Box 608
 Cheyenne, WY 82003
-0608
 Ph. 307.633.4
242
 Fax 307.633.4240
 Email: 
election@laramiecountyclerk.com
 925 Sage Avenue, Suite 101
 Kemmerer, WY 83101
 Ph. 307.877.
2027
 or 2021
 Fax 307.877.3101
 Email: 
jpainter@lcwy.org
              
abru
nski@lcwy.org
 Natrona County Clerk
 Niobrara County Clerk
 200 North Center Street
 P.O. Box 863
 Casper, WY 82601
-0863
 Ph. 307.235.9217
 Fax 307.235.9367
 Email: 
elections@natronacounty
-wy.gov
 P.O. Box 420
 Lusk, WY 82225
-0420
 Ph. 307.334.2211
 Fax 307.334.3013
 Email: 
bfreeman@niobraracounty.org
 Park County Clerk
 Platte County Clerk
 1002 Sheridan
 Cody, WY 82414
-3589 Ph. 307.527.86
20 Fax 307.527.8626
 Email: 
voterinfo@parkcounty.us
 P.O. Drawer 728
 Wheatland, WY 82201
-0728
 Ph. 307.322.2315
 Fax 307.322.2245
 Email: 
pcelection@plattecountywyoming.com
 
 Page 
3 of 
3  Sheridan County Clerk
 Sublette County Clerk
 224 South 
Main Street; Suite B
-17 Sheridan, WY 82801
 Ph. 307.674.2515
 Fax 307.674.2909
 Email: 
elections@sheridancounty.com
 P.O. Box 250
 Pinedale, WY 82941
-0250
 Ph. 307.367.4372
 Fax 307.367.6396
 Email: 
Carrie.Long@sublettewyo.com
 Sweetwater County Clerk
 Teton County Clerk
 80 West Flaming Gorge Way, Suite 155
 Green River, WY 82935
 Ph. 307.872.3733
 Fax 307.872.3994
 Email: 
elections@sweet.wy.us
 P.O. Box 1727
 Jackson, WY 83001
-1727
 Ph. 307.733.4430
 Fax 307.739.8681
 Email: 
elections@tetoncountywy.gov
 Uinta County Clerk
 Washakie County Clerk
 P.O. Box 810
 Evanston, WY 82931
 Ph. 307.783.0306
 Fax 307.783.0376
 Email: 
kigoble@uintacounty.com
 1001 Big Horn Avenue
 P.O. Box 260
 Worland, WY 82401
 Ph. 307.347.3131
 Fax 307.347.9366
 Email: 
clerk@washakiecounty.net
  Weston County Clerk
   1 West Main
 Newcastle, WY 82701
 Ph. 307.746.4744
 Fax 307.746.9505
 Email: 
clerk@westongov.com



Yes, it's doable, but the fact that the data is so intertwined will make it a huge pain.  I understand the principle of re-gathering the data regularly, but this data just seems like it'll be super consistent anyway when it changes.  Maybe we can just hold a copy of the raw text and diff it once a week to see if its changed?

Reply:
ok nevermind thinking about it more I see how that's just not gonna work.  I'll get it done as you described

Reply:
Don't go overboard making your code super-generalizable!  I suspect your parsing logic is fine.  It's just the logic around wiring up all the parts (downloading the files, linking them etc ...) that we want to automate.